dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/ConverterService.java

```diff
@@ -149,14 +149,29 @@ private Annotation getAnnotationForRestObject(BaseObjectRest restObject) {
         DSpaceRestRepository repositoryToUse = utils
             .getResourceRepositoryByCategoryAndModel(baseObjectRest.getCategory(), baseObjectRest.getType());
         Annotation preAuthorize = null;
+        int maxDepth = 0;
         for (Method m : repositoryToUse.getClass().getMethods()) {
             if (StringUtils.equalsIgnoreCase(m.getName(), "findOne")) {
-                preAuthorize = AnnotationUtils.findAnnotation(m, PreAuthorize.class);
+                int depth = howManySuperclass(m.getDeclaringClass());
```

Comment:
why is this customization necessary?

Reply:
this is a fix for https://jira.lyrasis.org/browse/DS-4530 that was initially discovery working on this PR

Reply:
Isn't the repositoryToUse supposed to make sure the correct method is found?
I also wouldn't see how this would solve a problem with retrieving the annotation since DSpaceRestRepository also has a superclass